From df0107659d87a25d4b170acb0f86e594a2d3c1cd Mon Sep 17 00:00:00 2001
From: Naoto Yamaguchi <i33399_YAMAGUCHI@aisin-aw.co.jp>
Date: Mon, 30 Dec 2019 02:15:27 +0900
Subject: [PATCH] Add gcov support

---
 configure | 13 ++++++++++++-
 1 file changed, 12 insertions(+), 1 deletion(-)

diff --git a/configure b/configure
index 6310e0b9..ab655fc9 100755
--- a/configure
+++ b/configure
@@ -30,6 +30,7 @@ System types:
 Optional features:
   --enable-optimize=...   optimize listed components for speed over size [auto]
   --enable-debug          build with debugging information [disabled]
+  --enable-gcov           build with gcov [disabled]
   --enable-warnings       build with recommended warnings flags [disabled]
   --enable-wrapper=...    build given musl toolchain wrapper [auto]
   --disable-shared        inhibit building shared library [enabled]
@@ -133,6 +134,7 @@ build=
 target=
 optimize=auto
 debug=no
+gcov=no
 warnings=no
 shared=auto
 static=yes
@@ -159,6 +161,8 @@ case "$arg" in
 --disable-optimize) optimize=no ;;
 --enable-debug|--enable-debug=yes) debug=yes ;;
 --disable-debug|--enable-debug=no) debug=no ;;
+--enable-gcov|--enable-gcov=yes) gcov=yes ;;
+--disable-gcov|--enable-gcov=no) gcov=no ;;
 --enable-warnings|--enable-warnings=yes) warnings=yes ;;
 --disable-warnings|--enable-warnings=no) warnings=no ;;
 --enable-wrapper|--enable-wrapper=yes) wrapper=detect ;;
@@ -388,6 +392,13 @@ tryflag CFLAGS_MEMOPS -fno-tree-loop-distribute-patterns
 #
 test "$debug" = yes && CFLAGS_AUTO=-g
 
+#
+# Enable gcov if requessted.
+#
+test "$gcov" = yes && CFLAGS_AUTO=--coverage
+test "$gcov" = yes && CXXFLAGS_AUTO=--coverage
+test "$gcov" = yes && LDFLAGS_AUTO=-lgcov
+
 #
 # Preprocess asm files to add extra debugging information if debug is
 # enabled, our assembler supports the needed directives, and the
@@ -734,7 +745,7 @@ CPPFLAGS = $CPPFLAGS
 LDFLAGS = $LDFLAGS
 LDFLAGS_AUTO = $LDFLAGS_AUTO
 CROSS_COMPILE = $CROSS_COMPILE
-LIBCC = $LIBCC
+LIBCC = $LIBCC $LIBCC_AUTO
 OPTIMIZE_GLOBS = $OPTIMIZE_GLOBS
 ALL_TOOLS = $tools
 TOOL_LIBS = $tool_libs
-- 
2.17.1

